adapters/celestia/src/verifier/address.rs

```diff
@@ -1,64 +1,116 @@
 use std::fmt::{Display, Formatter};
 use std::str::FromStr;
 
+use bech32::WriteBase32;
 use borsh::{BorshDeserialize, BorshSerialize};
 use serde::{Deserialize, Serialize};
 use sov_rollup_interface::AddressTrait;
 
+/// Human Readable Part: "celestia" for Celestia network
 const HRP: &str = "celestia";
 
+/// Representation of the address in the Celestia network
 #[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
-// Raw ASCII bytes, including HRP
-// TODO: https://github.com/Sovereign-Labs/sovereign-sdk/issues/469
-pub struct CelestiaAddress(Vec<u8>);
+pub struct CelestiaAddress {
+    // Storing vector of u5 each as u8
+    data: Vec<u8>,
```

Comment:
Celestia addresses are [u8;32] bech32 encoded. Can we use the `[u8;32]` as our internal representation the same way we do in our `Bech32Address` type?

Reply:
That's good point. I think I overgeneralized implementation. I will double check Celestia codebase to see which variant they use and how many bytes this it.

Reply:
Addressed this